src/neuroconv/tools/nwb_helpers/_configure_backend.py

```diff
@@ -0,0 +1,24 @@
+"""Collection of helper functions related to configuration of datasets dependent on backend."""
+from typing import Union
+
+from pynwb import NWBFile
+
+from ._configuration_models._hdf5_backend import HDF5BackendConfiguration
+from ._configuration_models._zarr_backend import ZarrBackendConfiguration
+
+
+def configure_backend(
```

Comment:
I woud like a more specific name here. Consistency with our previos convention would get us `configure_io_backend` I think. We could also have `configure_write_backend` or `set_write_backend` as that is mainly what this is doing.

Reply:
I am not sure about write it could be seomthing else but I guess the biggest thing is that I disagree with:

>  since Backend as a word makes it clearer it has to do with the storage structure on disk

Where are you getting this? Backend has multiple meanings and I am not sure that the one we are using here is the most common one. The most common use of Backend in SE is its use in the server-client model and most of the time is code for a database. There is also plotting backend. Point being that backend is not clear enough by itself, this is a backend for IO and at that we are only using it for writing (output) but I am fine with anything else. I am not that keen on write but I wanted to point out that Backend is the unprecise thing here.

Reply:
So you want

`BackendConfiguration` -> `BackendIOConfiguration`

and

`configure_backend` -> `configure_backend_io`

?

Reply:
Yes, that's good.

I prefer `configure_io_backend` but `IOBackendConfiguration` maybe not so good? I think your native language ear might be good at this. Both are good to me.